### PR TITLE
Feature/email sign in

### PR DESCRIPTION
**Pull request by chasm**
Changed the schema to eliminate passwords and create an email sign-in feature.

### CODE REVIEW

File: docs/sign-up-in-flow.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+SIGN UP
+  1. Delete all expired keys
+  2. Delete all unverified users without keys
+  3. If user:
+    1. EXISTS
+      1. Set user `key` IF NOT EXISTS
+      2. Set user `keyExpires` to 30 minutes from now
```
Comment (ullibodnar) [inline docs/sign-up-in-flow.txt]: Why 30 minutes? This seems like a short time for someone that is using a screen reader, has a disability, etc.

File: src/constants.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const THIRTY_MINUTES = 1000 * 60 * 30
```
Comment (ullibodnar) [inline src/constants.js]: Should we change this const to be "THIRTY_MINUTES_EXPIRATION_MULTIPLIER" or something of the sorts instead? Thirty minutes seems vague...

File: src/utilities/sendSignInEmail/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const link = `http://localhost:3000/complete-sign-in?id=${key}${
+    remainSignedIn ? '&rsi=true' : ''
+  }${up ? '&up=true' : ''}`
+
+  const message = {
+    to: email,
+    from: 'info@jsfest.nz',
+    subject: 'Sign in to JSFest.nz',
+    text: `
+Copy and paste this link into your browser to sign in to JSFest.nz:
```
Comment (ullibodnar) [inline src/utilities/sendSignInEmail/index.js]: "Open this link in your browser"?

File: src/constants.js (hunk 2 above)
Comment (chasm) [reply src/constants.js]: Switched to ONE_DAY_IN_MILLISECONDS.

File: docs/sign-up-in-flow.txt (hunk 1 above)
Comment (chasm) [reply docs/sign-up-in-flow.txt]: Changed to 1 day.

File: src/utilities/sendSignInEmail/index.js (hunk 3 above)
Comment (chasm) [reply src/utilities/sendSignInEmail/index.js]: This is in the text version of the email only. Many text-only email apps do not recognise such URLs and link them -- this is fallback only and it's likely that no one will ever see it. That said, I'll change it to "Open this link in your browser (copy and paste the URL if necessary)..."

**Pull request merged**